cpp/tests/datetime/datetime_ops_test.cu

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/datetime.hpp>
+#include <cudf/utilities/chrono.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/column/column_factories.hpp>
+
+#include <tests/utilities/type_lists.hpp>
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/timestamp_utilities.cuh>
+
+#include <tests/utilities/legacy/cudf_test_utils.cuh>
+
+#include <gmock/gmock.h>
+
+template <typename T>
+struct DatetimeOpsTest : public cudf::test::BaseFixture {
+  cudaStream_t stream() { return cudaStream_t(0); }
+  cudf::size_type size() { return cudf::size_type(10); }
+  cudf::data_type type() { return cudf::data_type{cudf::experimental::type_to_id<T>()}; }
+};
+
+template <typename Element>
+void print_column(cudf::column_view col) {
+  print_typed_column<Element>(
+    col.data<Element>(),
+    (gdf_valid_type*) col.null_mask(),
+    col.size(),
+    1);
+}
+
+TYPED_TEST_CASE(DatetimeOpsTest, cudf::test::TimestampTypes);
+
+TYPED_TEST(DatetimeOpsTest, TestExtractingDatetimeComponents) {
+
+  using namespace cudf::test;
+  using namespace simt::std::chrono;
+
+  auto test_timestamps_D = fixed_width_column_wrapper<cudf::timestamp_D>{
+    cudf::timestamp_D{-1528}, // 1965-10-26
+    cudf::timestamp_D{17716}, // 2018-07-04
+    cudf::timestamp_D{19382}, // 2023-01-25
+  };
+
+  auto test_timestamps_s = fixed_width_column_wrapper<cudf::timestamp_s>{
+		cudf::timestamp_s{-131968728}, // 1965-10-26 14:01:12
+		cudf::timestamp_s{1530705600}, // 2018-07-04 12:00:00
+		cudf::timestamp_s{1674631932}, // 2023-01-25 07:32:12
+	};
+
+  auto test_timestamps_ms = fixed_width_column_wrapper<cudf::timestamp_ms>{
+		cudf::timestamp_ms{-131968727238}, // 1965-10-26 14:01:12.762
+		cudf::timestamp_ms{1530705600000}, // 2018-07-04 12:00:00.000
+		cudf::timestamp_ms{1674631932929}, // 2023-01-25 07:32:12.929
+	};
+
+  auto test_timestamps_D_view = static_cast<cudf::column_view>(test_timestamps_D);
```

Comment:
This strictly isn't necessary. You could just pass the `fixed_width_column_wrapper` into the API and it will implicitly convert to `column_view`.